staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -82,6 +87,10 @@ type proxyHandlingInfo struct {
 	serviceAvailable bool
 	// servicePort is the port of the service this handler proxies to
 	servicePort int32
+	// defaultSNI is the default server name to use for TLS Server Name Indication (SNI)
+	defaultSNI string
+	// fallbackToDefaultSNI indicates whether to fallback to the defaultSNI instead of using an external SNI
+	fallbackToDefaultSNI bool
```

Comment:
not clear on why we'd have a conditional here.  I expect that the old behavior will remain forever.

Reply:
this was removed in the latest commit, but i originally added this to prevent extra requests for external services with certs that rely on the old behavior. if it failed once with a `x509.HostnameError`, it would default to the old behavior for all future requests.